src/components/Card.js

```diff
@@ -1,33 +1,40 @@
 export default class Card {
-  constructor({ name, link }, cardSelector, handleImageClick) {
-    this._name = name;
-    this._link = link;
+  constructor(
+    data,
+    cardSelector,
+    handleImageClick,
+    handleLikeClick,
+    handleDeleteClick
+  ) {
+    this._isLiked = data.isLiked;
+    this._id = data._id;
+    this._name = data.name;
+    this._link = data.link;
+    this._data = data;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleLikeClick = handleLikeClick;
+    this._handleDeleteClick = handleDeleteClick;
   }
 
   _setEventListeners() {
-    this._cardElement
-      .querySelector(".card__like-button")
-      .addEventListener("click", () => {
-        this._handleLikeIcon();
-      });
+    this._likeButton.addEventListener("click", () => {
+      this._handleLikeClick(this._data);
+    });
 
     this._cardElement
       .querySelector(".card__delete-button")
       .addEventListener("click", () => {
-        this._handleDeleteCard();
+        this._handleDeleteClick(this);
       });
 
     this._cardImageElement.addEventListener("click", () => {
       this._handleImageClick({ name: this._name, link: this._link });
     });
   }
 
-  _handleLikeIcon() {
-    this._cardElement
-      .querySelector(".card__like-button")
-      .classList.toggle("card__like-button_active");
+  toggleLike() {
+    this.this._likeButton.classList.toggle("card__like-button_active");
```

Comment:
* Likes don't work https://snipboard.io/mZMJbj.jpg 


`  this.this._likeButton`


It should be 

```js

    this._likeButton.classList.toggle("card__like-button_active");

```